indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx

```diff
@@ -158,36 +206,114 @@ function EditableListDisplay({editableList, codePresent, editableType, eventId})
 
   const toggleSelectAll = dataChecked => {
     if (dataChecked) {
-      setChecked(sortedList.map(row => row.id));
+      setChecked(editables.map(row => row.editable.id));
     } else {
       setChecked([]);
     }
   };
 
   const toggleSelectRow = dataIndex => {
-    const newRow = sortedList[dataIndex].id;
+    const newRow = sortedList[dataIndex].editable.id;
     if (checked.includes(newRow)) {
       setChecked(old => old.filter(row => row !== newRow));
     } else {
       setChecked(old => [...old, newRow]);
     }
   };
 
+  const checkedEditablesRequest = async (urlFunc, data = {}) => {
+    let response;
+    try {
+      response = await indicoAxios.post(urlFunc({confId: eventId, type: editableType}), {
+        editables: checked,
+        ...data,
+      });
+    } catch (error) {
+      handleAxiosError(error);
+      return null;
+    } finally {
+      setActiveRequest(null);
+    }
+
+    return camelizeKeys(response.data);
+  };
+
+  const downloadAllFiles = async () => {
+    setActiveRequest('download');
+    const rv = await checkedEditablesRequest(editablesArchiveURL);
+    if (rv) {
+      location.href = rv.downloadURL;
+    }
+  };
+
+  const assignEditor = async editor => {
```

Comment:
What do you think about merging the last three functions? Since we're repeating most of the codes perhaps we could move the `setActiveRequest` and `checkedEditablesRequest` arguments to a method.

Reply:
Yeah, even though I'd say it's right on the edge between where deduplicating makes sense or adds more complexity ;)